lib/puppet/property/list.rb

```diff
@@ -47,7 +47,7 @@ def retrieve
         #ok, some 'convention' if the list property is named groups, provider should implement a groups method
         tmp = provider.send(name) if provider
         if tmp && tmp != :absent
-          return tmp.split(delimiter)
+          return Array === tmp ? tmp : tmp.split(delimiter)
```

Comment:
Usually one would use `.instance_of?` for checking that `tmp` is strictly an object of Array type. 
`===` returns true also if `tmp` is an object of a subclass of Array.